### PR TITLE
Improvement/add users

### LINKED ISSUE #18: possibility to add new users

### PR DESCRIPTION
**Pull request by Borzygniew**
### Added a functionality to register new user

**Validation of:**
- email
- username
- password

Also created draft User class and change some variable names, initializations and declarations including good manners of latest C# version. Additionaly made shorter Logging() method.

_Windows UI is created by @goflyy and they gonna be reviewed in another pull request._

**MySQL:**

1. New procedure user_validation( [username], [email] )
2. User has his own email
3. Customer and barber foreign keys to user_id

#ROASTME

### CODE REVIEW

File: UberBarber/UberBarber/View/LoginView.xaml [hunk 1]
```diff
@@ -235,7 +235,44 @@
...
+                        </ControlTemplate>
+                    </Button.Template>
+                </Button>
+                <TextBlock
+                    Foreground="LightGray"
+                    FontSize="12"
+                    FontWeight="Medium"
+                    FontFamily="Montserrat"
+                    TextWrapping="Wrap"
+                               Margin="580,325,76,71" Grid.Row="1"><Run Language="pl-pl" Text="Don't have an account?"/></TextBlock>
```
Comment (goflyy) [inline UberBarber/UberBarber/View/LoginView.xaml:274]: I'll pretend I don't see it

File: UberBarber/UberBarber/View/UserRegistration.xaml.cs [hunk 2]
```diff
@@ -29,19 +31,27 @@ private void Window_MouseDown(object sender, MouseButtonEventArgs e)
...
         {
+            string username = text_username.Text;
+            string password = pswd_box.Password;
+            string confirm_password = pswd_box_confirm.Password;
+            string email = text_email.Text;
 
+            DatabaseQueries query = new DatabaseQueries();
+            query.Add_user(username, password, confirm_password, email);
+            Close();
+
```
Comment (goflyy) [inline UberBarber/UberBarber/View/UserRegistration.xaml.cs]: blank space

File: UberBarber/UberBarber/database/DatabaseQueries.cs [hunk 3]
```diff
@@ -11,45 +11,70 @@ namespace UberBarber.database
...
+            Open_connection();
+            MySqlCommand query = new($"SELECT username, password FROM serwer165956_projektstudia.user WHERE username = '{user_name}' AND password = md5('{user_password}');", _connection);
             try
             {
-                _database._connection.Open();
+                _reader = query.ExecuteReader();
+
+                if (!_reader.Read())
+                    MessageBox.Show("WRONG CREDENTIALS!");
+
```
Comment (goflyy) [inline UberBarber/UberBarber/database/DatabaseQueries.cs]: blank space

File: UberBarber/UberBarber/View/LoginView.xaml (hunk 1 above)
Comment (Borzygniew) [reply UberBarber/UberBarber/View/LoginView.xaml:274]: read pull request description

Comment (goflyy) [reply UberBarber/UberBarber/View/LoginView.xaml:274]: sorry i had to xD

File: UberBarber/UberBarber/View/UserRegistration.xaml.cs (hunk 2 above)
Comment (Borzygniew) [reply UberBarber/UberBarber/View/UserRegistration.xaml.cs]: deleted

File: UberBarber/UberBarber/database/DatabaseQueries.cs (hunk 3 above)
Comment (Borzygniew) [reply UberBarber/UberBarber/database/DatabaseQueries.cs]: deleted

**Review by goflyy — APPROVED** [review]

File: UberBarber/UberBarber/database/DbConnection.cs [hunk 4]
```diff
@@ -12,31 +14,45 @@ namespace UberBarber
...
+            catch (MySqlException)
+            {
+                MessageBox.Show("Cannot connect to server. Contact administrator");
+            }
         }
+
+        public void Close_connection()
+        {
+            // This method closes MySqlConnection.
+
```
Comment (Cialkowski) [inline UberBarber/UberBarber/database/DbConnection.cs:53]: add try except for this function simmilar to Open_connection

File: UberBarber/UberBarber/View/LoginView.xaml.cs [hunk 5]
```diff
@@ -25,31 +25,49 @@ public LoginView()
...
-            if (!query.logging(txtUser.Text, txtPassword.Password)) return;
-            new MainWindow().Show();
-            this.Close();
+            try
+            {
+                query.Logging(txtUser.Text, txtPassword.Password);
+                new MainWindow().Show();
+                Close();
+            }
+            catch (Exception ex)
```
Comment (Cialkowski) [inline UberBarber/UberBarber/View/LoginView.xaml.cs:62]: logging function is secured well, only thing here to catch is window opening. Specify exception type here

**Review by Cialkowski — CHANGES_REQUESTED** [review]

File: UberBarber/UberBarber/database/DbConnection.cs (hunk 4 above)
Comment (Borzygniew) [reply UberBarber/UberBarber/database/DbConnection.cs:53]: done

File: UberBarber/UberBarber/View/LoginView.xaml.cs (hunk 5 above)
Comment (Borzygniew) [reply UberBarber/UberBarber/View/LoginView.xaml.cs:62]: done

**Review by Cialkowski — APPROVED** [review]
Lgtm

**Pull request merged**